[Feature notification] Add EuiNotificationEvents

miukimiu opened the pull request:
### Summary

Provide a detailed summary of your PR. Explain how you arrived at your solution. If it includes changes to UI elements include a screenshot or gif.

### Checklist

- [ ] Check against **all themes** for compatibility in both light and dark modes
- [ ] Checked in **mobile**
- [ ] Checked in **Chrome**, **Safari**, **Edge**, and **Firefox**
- [ ] Props have proper **autodocs** and **[playground toggles](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#adding-playground-toggles)**
- [ ] Added **[documentation](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md)**
- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for the any docs examples
- [ ] Added or updated **[jest tests](https://github.com/elastic/eui/blob/master/wiki/testing.md)**
- [ ] Checked for **breaking changes** and labeled appropriately
- [ ] Checked for **accessibility** including keyboard-only and screenreader modes
- [ ] A **[changelog](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

miukimiu commented on src-docs/src/views/notification_events/notification_events.tsx:
```diff
@@ -0,0 +1,204 @@
+
+      <EuiContextMenuItem
+        key="contextMenuItemB"
+        onClick={() => onFilterByType(type)}>
```
@chandlerprall should we also consider undo the "view messages like this", like "View all type of messages"

chandlerprall replied:
I'd expect a button at the top of the notifications section to be present when a filter is active & removes that filter

miukimiu commented on src-docs/src/views/notification_events/notification_events.tsx:
```diff
@@ -0,0 +1,204 @@
+        onRead={onRead}
+        contextMenuItems={contextMenuItems}
+        onOpenContextMenu={onOpenContextMenu}
+        onClickPrimaryAction={onClickEventPrimaryAction}
+        onClickTitle={onClickNoNewsTitles!}
```
@chandlerprall I'm not also sure if this is the best way to handle the actions for the "title" and "primary action". Or if it is better to pass directly the actions on each `notificationEventsData` object.

chandlerprall replied:
Will take a specific look at this as well 👍

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

cchaos commented:
Btw, I've asked @miukimiu to do something similar to the EuiSelectableSitewideTemplate where it showcases which prop renders which UI element because it's not always clear just by the prop name.
<img width="968" alt="Screen Shot 2021-02-25 at 12 46 37 PM" src="https://user-images.githubusercontent.com/549577/109195040-0fc7f980-7768-11eb-9dbe-c4e128f4460a.png">

myasonik reviewed (approved):
👋 Hey, this looks great! I looked this over in browser but didn't have time to finish going through the code so I thought I'd leave the comments I had and I'll pick up again on Monday.

I'll probably have a few more little things like this, but I don't think there'll be anything big or surprising. 🎉

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

myasonik commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,218 @@
+  meta={{
+    type: 'Alert',
+    iconType: 'logoMaps',
+    eventName: 'alert-maps-01',
```
This prop, it seems, will be much less often used than `iconAriaLabel`. Maybe let's swap them?
```suggestion
    iconAriaLabel: 'Maps',
```

myasonik reviewed (changes requested):
A few disparate but little thing things. Let me know if you need help implementing or have any questions on these!

----

I'm getting a `Warning: Failed prop type: The prop 'meta.eventName
' is marked as required in 'EuiNotificationEven', but its value is 'undefined'.` error in the console when I load this PR locally. It doesn't seem to break anything and I couldn't track it down at a quick look, but we should probably fix these.

miukimiu commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,218 @@
+              A <strong>EuiNotificationEvent</strong> is comprised of different
+              props:
+            </p>
+            <ul>
```
Yes, it is very similar. I'm changing it to try to add something more similar to what @cchaos suggested:

> Btw, I've asked @miukimiu to do something similar to the EuiSelectableSitewideTemplate where it showcases which prop renders which UI element because it's not always clear just by the prop name.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

miukimiu commented:
Thanks, @myasonik for all the code suggestions. I went through your comments and fixed the a11y issues.

@cchaos I added a template that showcases which prop renders which UI element and I tried to explain how the prop should be used. Let me know what you think.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

myasonik reviewed (approved):
a11y looks good - thanks for all your work!

The guided demo portion could probably be improved upon a bit but the information is presented multiple ways so I'm not going to spend more time on it right now.

CI seems to be failing because the dev server is failing to start for the axe tests. _I think_ if you merge master, this has been improved and hopefully CI will feel less flaky. 

🚀

cchaos commented on src/components/notification/notification_event_meta.tsx:
```diff
@@ -107,14 +120,21 @@ export const EuiNotificationEventMeta: FunctionComponent<EuiNotificationEventMet
     ? { 'aria-label': iconAriaLabel }
     : { 'aria-hidden': true };
 
+  const onOpenPopover = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+    if (onOpenContextMenu) {
+      setContextMenuItems(onOpenContextMenu());
+    }
+  };
+
   return (
     <div className={classes}>
       <div className="euiNotificationEventMeta__section">
         {typeof isRead === 'boolean' && (
           <EuiNotificationEventReadButton
             isRead={isRead}
             onClick={onMarkAsRead}
-            eventName={eventName}
+            eventName={eventName!}
```
Also, I'd now differentiate between the string passed to `title` and `aria-label` of the read button. Before the tooltip was nice and succinct `Mark as read`, now since it matches the aria-label, it's not an easy read.

<img width="333" alt="Screen Shot 2021-03-02 at 19 00 00 PM" src="https://user-images.githubusercontent.com/549577/109731711-c9b8ce80-7b89-11eb-9f15-9ff8522ea018.png">

miukimiu replied:
The `eventName` used to be required in **EuiNotificationEventMeta**. But @myasonik suggested using the `title` to fill the `eventName` by default.

So this is the way I found it to make it work. Not sure it is the best way to handle it. But here's the explanation.

We want to pass the `title` from the parent `<EuiNotificationEvent />` to the children in case no `meta.eventName` is passed by the consumer:

```js
<EuiNotificationEventMeta
   eventName={meta.eventName ?? title}
/>
```

The `title` is required on the parent component and because of this, it will always exist. So the `eventName` on the children component `<EuiNotificationEventMeta />` is going to exist as well. So `eventName!`. 

But if I was making the `eventName` required, consumers would always have to pass the `eventName` in the `meta` object:

```js
<EuiNotificationEvent
  meta={{
    type: 'Report',
    iconType: 'logoKibana',
    iconAriaLabel: 'Kibana',
    time: '1 min ago',
    // this would be required
    eventName: "event-01"
  }}
/>
```

Also the `<EuiNotificationEventMeta />`  is not to meant to be used on its own by consumers.

miukimiu replied:
Maybe we can bring back the `eventName` as required. @myasonik what do you think? 

> Also, I'd now differentiate between the string passed to `title` and `aria-label` of the read button. Before the tooltip was nice and succinct `Mark as read`, now since it matches the aria-label, it's not an easy read.
> 
> <img alt="Screen Shot 2021-03-02 at 19 00 00 PM" width="333" src="https://user-images.githubusercontent.com/549577/109731711-c9b8ce80-7b89-11eb-9f15-9ff8522ea018.png">

cchaos replied:
Yep, I understand why you were using that format. However, I think we need to lean on props and not assume at certain points. This may be another good reason to spread out the meta props at the top level of `EuiNotificationEvent` instead of nesting.

myasonik replied:
Pinging our TS gurus for some help on this: @chandlerprall @thompsongl 

Making `eventName` required would be kind of unfortunate from the consumer perspective if that means consumers need to pass in the same string for two properties without a super obvious differentiator. 

If spreading out the props like @cchaos suggests fixes it, then I don't see a reason not to do it.

miukimiu replied:
Spreading out the meta props doesn't fix. Unless I'm missing something.

The only way I'm seeing to avoid making the `eventName` required  is:

* pass the `title` from `EuiNotificationEvent` to the `<EuiNotificationEventMeta title={title} />` component and replace all the `eventName` with `title`. We wouldn't need two props to do the same. But then consumers would lose to ability to pass something more unique for aria attributes.
 * in the parent component `<EuiNotificationEventMeta  eventName={meta.eventName ?? title} />` and in the children say that `eventName!`

But maybe our TS gurus (@chandlerprall @thompsongl) can recommend a better way of handling this. 

Also, we cannot forget that the purpose of the `eventName` is to be used in aria attributes. I think it would make more sense to make it **required** and it would solve these scenarios:

> <img alt="Screen Shot 2021-03-02 at 19 00 00 PM" width="333" src="https://user-images.githubusercontent.com/549577/109731711-c9b8ce80-7b89-11eb-9f15-9ff8522ea018.png">

I think is more friendly to use this type of event name than forcing the title. Or it isn't?  🤔 
<img width="326" alt="Screenshot 2021-03-03 at 17 36 1@2x" src="https://user-images.githubusercontent.com/2750668/109848126-d3087080-7c47-11eb-9be5-edec9e51b2b8.png">

cchaos replied:
I think at this point it would be easier to show with a PR because there may be misunderstandings. I'll throw one together.

cchaos replied:
Here's the PR. I added more explanation in the PR summary and review comments. But essentially, I agree with the question above as to why there's even a difference between `eventName` and `title`. But you can read more here: https://github.com/miukimiu/eui/pull/16

miukimiu replied:
Thanks @cchaos for the PR. It was really helpful. 🙏🏽

thompsongl replied:
> Pinging our TS gurus for some help on this

Still looking for help here or did Caroline's PR resolve it?

miukimiu replied:
Caroline's PR resolved it! 👍🏽

cchaos commented on src-docs/src/views/notification_event/props.tsx:
```diff
@@ -0,0 +1,7 @@
+
+import { EuiNotificationEventPrimaryActionProps } from '../../../../src/components/notification/notification_event';
+
+export const EuiNotificationEventPrimaryAction: FunctionComponent<EuiNotificationEventPrimaryActionProps> = () => (
```
I think you can simply use the `'!!prop-loader!` syntax directly in the `_example.js` file for this now.

ex:

`import { FieldValueToggleGroupFilterItemType } from '!prop-loader!../../../../../src/components/search_bar/filters/field_value_toggle_group_filter.tsx';`

miukimiu replied:
When I use the prop loader it looks like this:

<img width="966" alt="Screenshot 2021-03-03 at 12 10 00" src="https://user-images.githubusercontent.com/2750668/109804016-ae959f80-7c19-11eb-933e-a58ad410df27.png">

That's why I decided to create the `props.tsx` file:

<img width="964" alt="Screenshot 2021-03-03 at 12 11 28" src="https://user-images.githubusercontent.com/2750668/109804038-b5241700-7c19-11eb-8bc6-9cfb5c57533f.png">

cchaos replied:
Oh man, I didn't realize how badly the props loader did that. K!

cchaos commented on src-docs/src/views/notification_event/notifications_feed.tsx:
```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react';
```
Might want to consider turning these all to `js` files so the codesandbox links can work.

cchaos replied:
^^ @miukimiu I think you'll still want to do this. But it can happen right before merge so that any further updates to types would be caught in the examples. Just wanted to call it out again so it didn't get lost.

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,241 @@
+              <li>
+                <EuiCode>isRead</EuiCode>: Use this prop to show a button that
+                indicates the current <EuiCode>isRead</EuiCode> state of the
+                event. Use the <EuiCode>onRead</EuiCode> to allow users to
```
```suggestion
                event. Use <EuiCode>onRead</EuiCode> to allow users to
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,241 @@
+              </li>
+              <li>
+                <EuiCode>meta</EuiCode> (required): Provides important
+                information about the event, like when it was received, the type
```
```suggestion
                information about the event, like the <EuiCode>time</EuiCode> it was received, the <EuiCode>type</EuiCode>
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,241 @@
+                of event, and its severity. Use the{' '}
+                <EuiCode>badgeColor</EuiCode> in conjunction with the{' '}
+                <EuiCode>severity</EuiCode> to better indicate the level of
+                urgency. You can also pass an icon/logo to help users quickly
```
```suggestion
                urgency. You can also pass an <EuiCode>iconType</EuiCode> to help users quickly
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,241 @@
+              </li>
+              <li>
+                <EuiCode>onContextMenu</EuiCode>: Use this prop when you have
+                multiple events, and you need to add individual actions to each
```
```suggestion
                multiple events and you need to add individual actions to each
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,241 @@
+                <EuiCode>onContextMenu</EuiCode>: Use this prop when you have
+                multiple events, and you need to add individual actions to each
+                event. You can add filters based on the event type or a more
+                descriptive read/unread action as an alternative to the read
```
```suggestion
                descriptive read/unread actions as an alternative to the read
```

cchaos reviewed (changes requested):
This looks great. Thank you very much for the details examples. The Anatomy diagram is especially helpful! 🎉🎉🎉🎉

cchaos commented on src-docs/src/components/guide_components.scss:
```diff
@@ -241,6 +241,72 @@ $guideZLevelHighest: $euiZLevel9 + 1000;
 
+.guideDemo__notificationEventCircleIndicator {
+  display: inline-block;
+  width: 16px;
+  height: 16px;
```
As I then scanned the code, I realized you weren't actually using the component here, but a plain EuiPanel with classed divs. So might depend on how you want to handle that.

myasonik commented on src/components/notification/notification_event.tsx:
```diff
@@ -0,0 +1,168 @@
+  const titleProps = {
+    id: randomHeadingId,
+    className: classesTitle,
+    'data-test-subj': 'notificationEventTitle',
```
++

If configurable isn't possible, it at least needs to be unique and consistent otherwise this will bite us in Kibana at some point (maybe by attaching a stringified version of the `title` or something like that).

miukimiu replied:
I tried to add a unique `data-test-subj`  by concatenating the `id` at the end. For testing, I know that I'm concatenating with the `id`.

```
findTestSubject(component, 'notificationEventTitle-myId').simulate('click');
```
But how is the consumer going to know? 

Should I add this information to the props table? Or is there a better way to make this? 

cc @chandlerprall

myasonik replied:
Yeah, you can't use the random id you're generating because there's no way for the consumer to know that but if you use something the consumer is passing in already, then they can reconstruct it.

I don't think you really need to add a lot of docs around this because someone hunting for a DOM element to hook a test subject to will notice (I think). (At least that's been the case so far.)

chandlerprall replied:
I think the cleanest solution would be to add `titleProps` as a prop on the component itself. Defined either as the full set of `EuiLink` props, or a limited one with maybe only `onClick` & `data-test-subj`.

miukimiu replied:
@myasonik I was referring to concatenate the required `id` that is passed by the consumer in `<EuiNotificationEvent id={uniqueId} />`

I talked with @chandlerprall about this issue and agreed to prepend the `id` like `${id}-notificationEventTitle`. We also have more `data-test-subj` so I prepended the `id` for all of them.

myasonik commented on src/components/notification/notification_event_meta.tsx:
```diff
@@ -157,14 +177,16 @@ export const EuiNotificationEventMeta: FunctionComponent<EuiNotificationEventMet
             }
             closePopover={() => setIsPopoverOpen(false)}>
-            <EuiContextMenuPanel items={contextMenuItems} />
+            <div onClick={() => setIsPopoverOpen(false)}>
```
++

Good catch! Definitely can't add `onClick` on a `div`.

miukimiu replied:
@chandler introduced as a way to close the **EuiContextMenu** after an item is clicked. Here's the explanation: https://github.com/miukimiu/eui/pull/14#issue-576999056.

cchaos replied:
It would be great to start leaving code comments in here to explain some of these patterns. If there absolutely is a necessity to this, and doesn't hinder a11y, then a comment would be helpful to future people looking at this file.

chandlerprall replied:
The purpose of this onClick is to catch the event as it bubbles, not that the div itself is necessarily clickable. I don't think there's another way to know when the context menu was interacted with without being very invasive into `contextMenuItems` or being very restrictive about how an item can be defined.

cchaos replied:
I'm just wondering what happens if this is a multi-panel context menu, like clicking one of the items pushes to a new popover panel content. Is this going to disallow that because it will always force close the popover?

chandlerprall replied:
Good catch - it might, but I think the panel navigation logic should prevent bubbling in that case as the click was handled. Our event handling tends to lack in that area. Open to other ideas why this is bad, or how to implement here

cc @thompsongl for any ideas

cchaos replied:
We can leave this in for now and see how it affects consumer usages. It doesn't seem like something that would be a breaking change if it needs to be removed/changed.

myasonik commented on src/components/notification/notification_event.tsx:
```diff
@@ -0,0 +1,168 @@
+};
+
+export type EuiNotificationHeadingLevel =
+  | 'h1'
```
Most likely you're right but it seems odd to ban it at the same time, don't you think? Like there's no reason it theoretically couldn't be...

cchaos replied:
Because we are already manually listing the heading levels, I think it's a good opportunity to establish good patterns. Using an `<h1>` is not how this component is intended to be used. It can't be a page header.

myasonik commented on src/components/notification/notification_event.tsx:
```diff
@@ -0,0 +1,168 @@
+  /**
+   * The title of the event.
+   */
+  title: string;
```
This would add quite a bit of complexity as we pass the `title` into a bunch of `aria-*` attributes. How definitely do you want this? (I'm thinking it's easier to make it more lax in the future, rather than stricter.) 

We've worked around similar problems like this by reading the text of the rendered DOM but doing that has some trade-offs, namely: a performance hit and some a potential area for confusion.

The performance hit is usually small but I wonder if it might become an issue when rendering a potentially large list of notifications. (@chandlerprall or @thompsongl - do either of you know how significant the impact of [InnerText](https://eui.elastic.co/#/utilities/inner-text) is? I just remeber Chandler mentioning this to me at one point.)

And the potential for confusion is dependent on what sort of structured content someone might pass into this. The more complicated the `ReactNode` is, the more likely the text is difficult-to-impossible to understand when it's flattened into a plain string.

cchaos replied:
That's fine if we want to hold on this and open up to more flexibility in the future.

miukimiu commented on src/components/index.js:
```diff
@@ -241,7 +241,10 @@ export {
 
-export { EuiNotificationEventReadButton } from './notification';
+export {
+  EuiNotificationEventReadButton,
```
I was exporting this one because it could then be used inside the Notifications Center page. But is it a good idea? Also, the playground is so lonely with only this component... 😞  

What do you think. Should I stop export it?

cchaos replied:
Yeah I'd just remove it from the exports for now until we have more alignment on what the Center page is going to need.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

miukimiu commented:
Based on the PR reviews:

* Tests and docs usage were updated to reflect the `EventName` change.
* Updated the **more messages button**. Now it says **+ 2 more** and the link is using the primary color. If there are only 2 messages the button doesn't appear. I think it was weird to add just one message to the accordion. The icon is now properly aligned with the bottom content.

<img width="590" alt="Screenshot 2021-03-04 at 15 49 25" src="https://user-images.githubusercontent.com/2750668/109990363-3acec200-7d01-11eb-9c4d-4778629b2821.png">

* Fixed the demo to make it responsive (badge truncates) and to reflect the meta props that are now spread out:

<img width="289" alt="Screenshot 2021-03-04 at 15 51 03" src="https://user-images.githubusercontent.com/2750668/109990607-78334f80-7d01-11eb-8963-e34019568b77.png">


* No more playground for the **EuiNotificationEventReadButton** component and it is no longer exported.
* I did a few structural changes. The **EuiNotificationEventReadButton** is no longer part of the meta component.
* I also tried different usage scenarios. With and without the read button and the context menu.

<img width="1162" alt="tests@2x" src="https://user-images.githubusercontent.com/2750668/109992962-b6ca0980-7d03-11eb-9a96-93bda718c40b.png">

* Split the `primaryAction` prop into `primaryAction` and `primaryActionProps`

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+          <EuiText>
+            <p>
+              The <strong>EuiNotificationEvent</strong> takes into account that
+              an event can be purely informative or actionable. It&apos;s
```
```suggestion
              an event can be purely informative or actionable. It is
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+            <p>
+              The <strong>EuiNotificationEvent</strong> takes into account that
+              an event can be purely informative or actionable. It&apos;s
+              flexible enough and adapts the design according to the props
```
```suggestion
              flexible and adapts the design according to the props
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+          <EuiText>
+            <ul style={{ listStyleType: 'upper-alpha' }}>
+              <li>
+                <EuiCode>isRead</EuiCode>: Use this prop to show a button that
+                indicates the current <EuiCode>isRead</EuiCode> state of the
+                event. Use <EuiCode>onRead</EuiCode> to allow users to toggle
+                between read and unread states.
```
```suggestion
                <EuiCode>isRead</EuiCode>: Shows a button that
                indicates the current <EuiCode>isRead</EuiCode> state of the
                event. Use <EuiCode>onRead</EuiCode> to allow users to toggle
                between read and unread states.
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+                between read and unread states.
+              </li>
+              <li>
+                <EuiCode>iconType</EuiCode> An <EuiCode>iconType</EuiCode> to
+                help users quickly identify an event.
```
```suggestion
                <EuiCode>iconType</EuiCode>: Display an icon or logo to
                help users quickly identify where the event originated.
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+                help users quickly identify an event.
+              </li>
+              <li>
+                <EuiCode>type</EuiCode> (required): Use this prop to show a
+                badge with the <EuiCode>type</EuiCode> of event and its{' '}
+                <EuiCode>severity</EuiCode>. Use the{' '}
+                <EuiCode>badgeColor</EuiCode> in conjunction with the{' '}
+                <EuiCode>severity</EuiCode> to better indicate the level of
+                urgency.
```
```suggestion
                <EuiCode>type</EuiCode> (required): Shows inside a
                badge denoting what type of event it is. Use in conjunction with{' '} 
                <EuiCode>severity</EuiCode> and <EuiCode>badgeColor</EuiCode>{' '}
                to indicate the level of urgency.
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+                urgency.
+              </li>
+              <li>
+                <EuiCode>time</EuiCode> (required): Use this prop to indicate
+                the time the event was received.
```
```suggestion
                <EuiCode>time</EuiCode> (required): Indicates
                the time the event was received. It is recommended 
                to display a relative time format like '2 hours ago'.
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+                about the notification.
+              </li>
+              <li>
+                <EuiCode>messages</EuiCode>: Use this prop to provide more
+                details about the event. You can provide just one message or
+                multiple messages when the event executes in various steps.
```
```suggestion
                <EuiCode>messages</EuiCode>: Provides more
                details about the event. You can provide a single message or
                multiple messages if the event executes in various steps.
```

cchaos commented on src-docs/src/views/notification_event/notification_event_example.js:
```diff
@@ -0,0 +1,248 @@
+          <EuiText>
+            <p>
+              The following demo shows how you can combine different props to
+              create different types of events like a report, alert, or even
```
```suggestion
              create different types of events like a report, alert, or simply
```

cchaos commented on src/components/notification/_notification_event_messages.scss:
```diff
@@ -0,0 +1,18 @@
+.euiNotificationEventMessages {
+  color: $euiColorDarkShade;
+  font-size: $euiFontSizeS;
```
Is this necessary since you have them wrapped in EuiText blocks? Also EuiText accepts `color="subdued"` for the lighter text color.

cchaos reviewed (commented):
🎉 Awesome, thank you for those updates @miukimiu !

I'm just seeing a couple more nit-picky design things.

1. I think because you've pulled the button out of the meta component, which established the overall min height the rest of the meta stuff aligned well vertically. Now not so much.  Maybe that container just needs a min-height of `24px` (the same as the size of the button).
<img width="594" alt="Screen Shot 2021-03-04 at 14 02 28 PM" src="https://user-images.githubusercontent.com/549577/110016049-92b1fc80-7cf2-11eb-9215-83b095505f25.png">

2. The only thing about showing the accordion only if there are more than 2 messages is the inconsistency of showing only 1 message when there are 3+. It's much better to have a consistent understanding that only the first message is ever shown, even if it says "+1 more".

---

The rest of my comments are mainly grammatical. But I really love that the event itself has all the required props at the top level now. It'll be so much easier to implement.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_4531/

cchaos reviewed (commented):
Looks great to me! Thank you for making all those changes. The docs are really helpful in explaining the correct usages of this component. 🎉
